test/e2e/clusterctl_upgrade_test.go

```diff
@@ -158,6 +226,18 @@ var _ = Describe("When testing clusterctl upgrades using ClusterClass (v1.3=>cur
 			InitWithProvidersContract: "v1beta1",
 			InitWithKubernetesVersion: "v1.26.0",
 			WorkloadFlavor:            "topology",
+			// This check ensures that ownerReference apiVersions are updated for all types after the upgrade.
+			// TODO: Extend this test to include all ClusterClass templates when v1beta2 is released.
```

Comment:
Just thought about it again and saw your code. This will just work automatically, right?

We already validate the ownerRefs of KubeadmControlPlaneTemplate and DockerClusterTemplate. The only reason we don't validate the upgrade is that there is no way to use a ClusterClass with an old apiVersion of those.

But I think this will happen naturally. (Just like we test v1.2 / v1.3 upgrades with ClusterClass)

So fine for me to drop the TODOs for that